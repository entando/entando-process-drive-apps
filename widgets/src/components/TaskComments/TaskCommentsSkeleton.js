import React from 'react';
import PropTypes from 'prop-types';
import withStyles from '@material-ui/core/styles/withStyles';
import Skeleton from '@material-ui/lab/Skeleton';
import Divider from '@material-ui/core/Divider';

const styles = {
  divider: {
    marginTop: '15px',
    marginBottom: '20px',
  },
  comment: {
    display: 'flex',
    marginTop: '10px',
    marginBottom: '5px',
    paddingLeft: '10px',
  },
  avatar: {
    margin: '5px 25px 5px 30px',
  },
  commentText: {
    marginTop: '5px',
  },
  commentInfo: {
    marginLeft: '15px',
    marginTop: '10px',
  },
  addComment: {
    marginTop: '30px',
  },
  textField: {
    marginTop: '40px',
  },
  controls: {
    display: 'flex',
    justifyContent: 'flex-end',
    marginTop: '20px',
  },
};

const GeneralInformationSkeleton = ({ classes }) => (
  <>
    <div>
      <Skeleton width="190px" height={20} variant="rect" />
      <Divider className={classes.divider} />
    </div>
    <div>
      <div className={classes.comment}>
        <Skeleton className={classes.avatar} width="40px" height="40px" variant="circle" />
        <div>
          <Skeleton className={classes.commentText} width="90px" height={15} variant="rect" />
          <Skeleton className={classes.commentInfo} width="190px" height={12} variant="rect" />
        </div>
      </div>
      <div className={classes.comment}>
        <Skeleton className={classes.avatar} width="40px" height="40px" variant="circle" />
        <div>
          <Skeleton className={classes.commentText} width="220px" height={15} variant="rect" />
          <Skeleton className={classes.commentInfo} width="190px" height={12} variant="rect" />
        </div>
      </div>
      <div className={classes.comment}>
        <Skeleton className={classes.avatar} width="40px" height="40px" variant="circle" />
        <div>
          <Skeleton className={classes.commentText} width="320px" height={15} variant="rect" />
          <Skeleton className={classes.commentInfo} width="190px" height={12} variant="rect" />
        </div>
      </div>
      <div className={classes.addComment}>
        <Skeleton width="50px" variant="rect" />
        <Divider className={classes.textField} />
      </div>
      <div className={classes.controls}>
        <Skeleton width="70px" height={30} variant="rect" />
      </div>
    </div>
  </>
);

GeneralInformationSkeleton.propTypes = {
  classes: PropTypes.shape({
    divider: PropTypes.string,
    comment: PropTypes.string,
    avatar: PropTypes.string,
    commentText: PropTypes.string,
    commentInfo: PropTypes.string,
    addComment: PropTypes.string,
    textField: PropTypes.string,
    controls: PropTypes.string,
  }).isRequired,
};

export default withStyles(styles)(GeneralInformationSkeleton);
