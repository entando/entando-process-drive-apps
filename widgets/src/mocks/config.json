{
  "payload": {
    "code": "pda_task_list",
    "config": {
      "groups": "[{\"label\":\"Administrators\",\"key\":\"Administrators\",\"checked\":true},{\"label\":\"broker\",\"key\":\"broker\",\"checked\":true},{\"label\":\"admin\",\"key\":\"admin\",\"checked\":false},{\"label\":\"rest-all\",\"key\":\"rest-all\",\"checked\":false},{\"label\":\"kiemgmt\",\"key\":\"kiemgmt\",\"checked\":true},{\"label\":\"kie-server\",\"key\":\"kie-server\",\"checked\":true},{\"label\":\"appraiser\",\"key\":\"appraiser\",\"checked\":false},{\"label\":\"manager\",\"key\":\"manager\",\"checked\":true},{\"label\":\"supplier\",\"key\":\"supplier\",\"checked\":true},{\"label\":\"IT\",\"key\":\"IT\",\"checked\":true},{\"label\":\"PM\",\"key\":\"PM\",\"checked\":true}]",
      "options": "[{\"key\":\"newPageOnClick\",\"label\":\"Open new page on table row click\",\"checked\":false},{\"key\":\"showClaim\",\"label\":\"Show Claim Button\",\"checked\":true},{\"key\":\"showComplete\",\"label\":\"Show Complete Button\",\"checked\":true}]",
      "columns": "[{\"name\":\"task-is-skipable\",\"position\":0,\"isVisible\":false},{\"name\":\"task-parent-id\",\"position\":1,\"isVisible\":false},{\"name\":\"task-created-by\",\"position\":2,\"isVisible\":true},{\"name\":\"task-input-data.application.property.saleprice\",\"position\":3,\"isVisible\":true},{\"name\":\"task-input-data.TaskName\",\"position\":4,\"isVisible\":true},{\"name\":\"initiator\",\"position\":5,\"isVisible\":true},{\"name\":\"task-created-on\",\"position\":6,\"isVisible\":true},{\"name\":\"task-priority\",\"position\":7,\"isVisible\":false},{\"name\":\"task-proc-def-id\",\"position\":8,\"isVisible\":true},{\"name\":\"task-form\",\"position\":9,\"isVisible\":true},{\"name\":\"task-skippable\",\"position\":10,\"isVisible\":true},{\"name\":\"task-input-data.application.applicant.name\",\"position\":11,\"isVisible\":true},{\"name\":\"task-actual-owner\",\"position\":12,\"isVisible\":true},{\"name\":\"task-activation-time\",\"position\":13,\"isVisible\":true},{\"name\":\"task-input-data.application.applicant.address\",\"position\":14,\"isVisible\":true},{\"name\":\"task-input-data.application.errors\",\"position\":15,\"isVisible\":true},{\"name\":\"task-name\",\"position\":16,\"isVisible\":true},{\"name\":\"task-input-data.application.property.address\",\"position\":17,\"isVisible\":true},{\"name\":\"task-subject\",\"position\":18,\"isVisible\":true},{\"name\":\"containerId\",\"position\":19,\"isVisible\":true},{\"name\":\"task-container-id\",\"position\":20,\"isVisible\":true},{\"name\":\"processDefinitionId\",\"position\":21,\"isVisible\":true},{\"name\":\"task-input-data.application.applicant.ssn\",\"position\":22,\"isVisible\":true},{\"name\":\"task-description\",\"position\":23,\"isVisible\":true},{\"name\":\"task-input-data.Skippable\",\"position\":24,\"isVisible\":true},{\"name\":\"task-workitem-id\",\"position\":25,\"isVisible\":true},{\"name\":\"task-expiration-time\",\"position\":26,\"isVisible\":true},{\"name\":\"task-input-data.application.property.age\",\"position\":27,\"isVisible\":true},{\"name\":\"task-status\",\"position\":28,\"isVisible\":true},{\"name\":\"task-input-data.application.mortgageamount\",\"position\":29,\"isVisible\":true},{\"name\":\"task-process-instance-id\",\"position\":30,\"isVisible\":true},{\"name\":\"task-input-data.ActorId\",\"position\":31,\"isVisible\":true},{\"name\":\"task-process-id\",\"position\":32,\"isVisible\":true},{\"name\":\"task-input-data.application.property.locale\",\"position\":33,\"isVisible\":true},{\"name\":\"task-input-data.application.amortization\",\"position\":34,\"isVisible\":true},{\"name\":\"task-input-data.application.applicant.creditrating\",\"position\":35,\"isVisible\":true},{\"name\":\"application\",\"position\":36,\"isVisible\":true},{\"name\":\"task-input-data.NodeName\",\"position\":37,\"isVisible\":true},{\"name\":\"task-proc-inst-id\",\"position\":38,\"isVisible\":true},{\"name\":\"task-input-data.application.downpayment\",\"position\":39,\"isVisible\":true},{\"name\":\"task-input-data.application.applicant.annualincome\",\"position\":40,\"isVisible\":true},{\"name\":\"task-id\",\"position\":41,\"isVisible\":true}]",
      "knowledgeSource": "kieStaging",
      "process": "itorders.orderhardware@itorders_1.0.0-SNAPSHOT"
    }
  },
  "metaData": {
    "status": "draft"
  },
  "errors": []
}
