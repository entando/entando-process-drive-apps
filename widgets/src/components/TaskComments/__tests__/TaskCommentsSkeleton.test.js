import { render } from '@testing-library/react';
import React from 'react';
import 'mocks/i18nMock';

import TaskCommentsSkeleton from 'components/TaskComments/TaskCommentsSkeleton';

describe('<TaskCommentsSkeleton />', () => {
  it('renders snapshot correctly', async () => {
    const { container } = render(<TaskCommentsSkeleton />);

    expect(container).toMatchSnapshot();
  });
});
