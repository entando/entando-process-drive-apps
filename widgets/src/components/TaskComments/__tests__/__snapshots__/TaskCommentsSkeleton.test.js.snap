// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TaskCommentsSkeleton /> renders snapshot correctly 1`] = `
<div>
  <div>
    <div
      class="MuiSkeleton-root MuiSkeleton-rect MuiSkeleton-animate"
      style="width: 190px; height: 20px;"
    />
    <hr
      class="MuiDivider-root GeneralInformationSkeleton-divider-1"
    />
  </div>
  <div>
    <div
      class="GeneralInformationSkeleton-comment-2"
    >
      <div
        class="MuiSkeleton-root MuiSkeleton-circle GeneralInformationSkeleton-avatar-3 MuiSkeleton-animate"
        style="width: 40px; height: 40px;"
      />
      <div>
        <div
          class="MuiSkeleton-root MuiSkeleton-rect GeneralInformationSkeleton-commentText-4 MuiSkeleton-animate"
          style="width: 90px; height: 15px;"
        />
        <div
          class="MuiSkeleton-root MuiSkeleton-rect GeneralInformationSkeleton-commentInfo-5 MuiSkeleton-animate"
          style="width: 190px; height: 12px;"
        />
      </div>
    </div>
    <div
      class="GeneralInformationSkeleton-comment-2"
    >
      <div
        class="MuiSkeleton-root MuiSkeleton-circle GeneralInformationSkeleton-avatar-3 MuiSkeleton-animate"
        style="width: 40px; height: 40px;"
      />
      <div>
        <div
          class="MuiSkeleton-root MuiSkeleton-rect GeneralInformationSkeleton-commentText-4 MuiSkeleton-animate"
          style="width: 220px; height: 15px;"
        />
        <div
          class="MuiSkeleton-root MuiSkeleton-rect GeneralInformationSkeleton-commentInfo-5 MuiSkeleton-animate"
          style="width: 190px; height: 12px;"
        />
      </div>
    </div>
    <div
      class="GeneralInformationSkeleton-comment-2"
    >
      <div
        class="MuiSkeleton-root MuiSkeleton-circle GeneralInformationSkeleton-avatar-3 MuiSkeleton-animate"
        style="width: 40px; height: 40px;"
      />
      <div>
        <div
          class="MuiSkeleton-root MuiSkeleton-rect GeneralInformationSkeleton-commentText-4 MuiSkeleton-animate"
          style="width: 320px; height: 15px;"
        />
        <div
          class="MuiSkeleton-root MuiSkeleton-rect GeneralInformationSkeleton-commentInfo-5 MuiSkeleton-animate"
          style="width: 190px; height: 12px;"
        />
      </div>
    </div>
    <div
      class="GeneralInformationSkeleton-addComment-6"
    >
      <div
        class="MuiSkeleton-root MuiSkeleton-rect MuiSkeleton-animate"
        style="width: 50px;"
      />
      <hr
        class="MuiDivider-root GeneralInformationSkeleton-textField-7"
      />
    </div>
    <div
      class="GeneralInformationSkeleton-controls-8"
    >
      <div
        class="MuiSkeleton-root MuiSkeleton-rect MuiSkeleton-animate"
        style="width: 70px; height: 30px;"
      />
    </div>
  </div>
</div>
`;
